nexus/db-queries/src/db/pool.rs

```diff
@@ -85,81 +88,60 @@ fn make_postgres_connector(
+        Self { log, connect_with, policy: None, collect_backtraces: None }
     }
 
-    /// Creates a new qorb-backed connection pool to a single instance of the
-    /// database.
-    ///
-    /// This is intended for tests that want to skip DNS resolution, relying
-    /// on a single instance of the database.
-    ///
-    /// In production, [Self::new] should be preferred.
-    pub fn new_single_host(log: &Logger, db_config: &DbConfig) -> Self {
-        let resolver = make_single_host_resolver(db_config);
-        let connector = make_postgres_connector(log);
-        let policy = Policy::default();
-        let inner = match qorb::pool::Pool::new(
-            "crdb-single-host".to_string(),
-            resolver,
-            connector,
-            policy,
-        ) {
-            Ok(pool) => {
-                debug!(log, "registered USDT probes");
-                pool
+    /// Connects to a single hard-coded database node.
+    pub fn new_single_host(log: &'a Logger, config: DbConfig) -> Self {
+        let connect_with = ConnectWith::SingleHost(Box::new(config));
+        Self { log, connect_with, policy: None, collect_backtraces: None }
+    }
+
+    pub fn policy(mut self, policy: Policy) -> Self {
+        self.policy = Some(policy);
+        self
+    }
+
+    pub fn collect_backtraces(mut self, collect_backtraces: bool) -> Self {
+        self.collect_backtraces = Some(collect_backtraces);
+        self
+    }
+
+    pub fn build(self) -> Pool {
+        let Self { log, connect_with, policy, collect_backtraces } = self;
+
+        let (resolver, name) = match connect_with {
+            ConnectWith::Resolver(resolver) => {
+                (resolver.for_service(ServiceName::Cockroach), "crdb")
             }
-            Err(err) => {
-                error!(log, "failed to register USDT probes");
-                err.into_inner()
+            ConnectWith::SingleHost(config) => {
+                (make_single_host_resolver(*config), "crdb-single-host")
             }
         };
-        Self::new_common(inner, log.clone())
-    }
-
-    /// Creates a new qorb-backed connection pool which returns an error
-    /// if claims are not available within one millisecond.
-    ///
-    /// This is intended for test-only usage, in particular for tests where
-    /// claim requests should rapidly return errors when a backend has been
-    /// intentionally disabled.
-    #[cfg(any(test, feature = "testing"))]
-    pub fn new_single_host_failfast(
-        log: &Logger,
-        db_config: &DbConfig,
-    ) -> Self {
-        let resolver = make_single_host_resolver(db_config);
         let connector = make_postgres_connector(log);
-        let policy = Policy {
-            claim_timeout: tokio::time::Duration::from_millis(1),
-            ..Default::default()
-        };
+
+        let policy = policy.unwrap_or_else(|| Policy::default());
+        let collect_backtraces = collect_backtraces.unwrap_or(true);
```

Comment:
This is where we match the current behavior for backtrace collection by default: unless requested otherwise, `collect_backtraces = true`.